packages/flutter/test/widget/inherited_test.dart

```diff
@@ -437,4 +460,16 @@ void main() {
     );
     expect(buildCount, equals(2));
   });
+
+  testWidgets('initState() dependency on Inherited asserts', (WidgetTester tester) async {
+    // This is a regression test for https://github.com/flutter/flutter/issues/5491
+    bool exceptionCaught = false;
+
+    TestInherited parent = new TestInherited(child: new ExpectFail(() {
+      exceptionCaught = true;
+    }));
+    await tester.pumpWidget(parent);
+
+    expect(exceptionCaught, isTrue);
```

Comment:
fwiw you could also use the widget tester's built-in "takeException" API to do this.